.github/workflows/docs.yaml

```diff
@@ -0,0 +1,57 @@
+name: Build and upload documentation
+
+"on":
+  merge_group: {}
+  pull_request: {}
+  push:
```

Comment:
On the DM side we usually do `branches: main` rather than try to exclude, but this is fine.

Reply:
This was copied from a SQuaRE template. I have no preference, and this can be changed if you like.